### PR TITLE
LPS-115056 Make bars dynamic

### PR DESCRIPTION
**Pull request by dgarciasarai**
The first version of the Audit Graph (mock data) 😄

audit-graph-mocked

cc: @jbalsas @wincent

### CODE REVIEW

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		</>
+	);
+}
+
+function CustomTick(props) {
+	const {payload, x, y} = props;
+
+	return (
+		<Text
+			fill="#666"
```
[the remark this answers is not in the document]
Comment (dgarciasarai) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js:98]: Sure!

**Comment by dgarciasarai** [comment]
Hey @bryceosterhaus, thanks for your comments :)

> I can imagine this might be an issue. We might have to manually handle this by changing the orientation prop on the YAxis and also reversed on the XAxis. This would swap the direction the data flows and which side the YAxis is on.

If RTL is activated in portal, we should pass this information in `props`? Like `isRtlEnabled` or something like that? I've never seen that. Is there an example? Let's investigate this a little bit and look for the best solution. This graph will be feature flagged for now.

> I think it might be how the logic with the ResponsiveContainer works. I just tested locally and if I snapshot a non-responsive container, It renders a full svg. You might try out running a timer, or perhaps even investigate the tests that Recharts has and how they test responsive containers.

I'll investigate a little bit more this because I want to add tests for the components :) I agree that this will be related to `ResponsiveContainer` because when we implemented the chart for Content Performance we have a full svg.

I'll update the pull request with the little changes (and hopefully with a test) and rerun the CI.

Thanks! 😄

**Comment by dgarciasarai** [comment]
More information about **testing**: these are the tests for `ResponsiveContainer` and `BarChart`
* https://github.com/recharts/recharts/blob/master/test/specs/component/ResponsiveContainerSpec.js
* https://github.com/recharts/recharts/blob/master/test/specs/chart/BarChartSpec.js

They test it both isolated. I didn't find any example testing both so I opened an issue: https://github.com/recharts/recharts/issues/2166. Let's see what is the answer for that. A workaround would be to build the component without the `ResponsiveContainer` and wrap it in the parent component. I'm not excited about that solution but it will fix our problem and let us test our components.

I'm going to update the pull request and run the CI.

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
_(I remove this from the parent description to not send it with the forward)_

The image above is the actual chart in `master`. There is a couple of things to think about it:
*  I let the vertical lines of the chart because with `ResponseContainer` I cannot place the end line to make a perfect react (only the edges). I tried putting `verticalPoints={['99%']}` but this attribute has to be an array of numbers (see http://recharts.org/en-US/api/CartesianGrid#verticalPoints). Also, @bryceosterhaus found this issue and write a new comment so we are waiting for a response: https://github.com/recharts/recharts/issues/1040. Just to clarify, the design doesn't have the vertical lines, only the edges.
* I found that the chart doesn't work fine with RTL direction. I don't know if there is something OTB for that in Recharts (see https://github.com/recharts/recharts/issues/263).
* For now, I didn't add the labels for the axis. I want to ask our designer because her proposal was different and I think we should double-check that.
* Testing components: I tried to test this `AuditBarChart` as I did with `EmptyAuditBarChart` with no results. Is this because Recharts renders an `svg`? How can we test it? This is what jest renders:
```
<body>
  <div>
    <div
      class="recharts-responsive-container"
      style="width: 100%; height: 400px;"
    >
      <div
        style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
      />
    </div>
  </div>
</body>
```

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:forward

**Pull request closed without merging**